libs/components/datetime/src/lib/modules/datepicker/daypicker-cell.component.ts

```diff
@@ -27,124 +27,137 @@ export class SkyDayPickerCellComponent implements OnInit, OnDestroy {
-      this.date.keyDate &&
-      this.date.keyDateText &&
+      !!this.date &&
+      !!this.date.keyDate &&
+      !!this.date.keyDateText &&
       this.date.keyDateText.length > 0 &&
       this.date.keyDateText[0].length > 0;
 
     // show the tooltip if this is the active date and is not the
     // initial active date (activeDateHasChanged)
     if (
-      this.datepicker.isActive(this.date) &&
+      !!this.date &&
+      this.#datepicker.isActive(this.date) &&
       this.activeDateHasChanged &&
       this.hasTooltip
     ) {
-      this.activeUid = this.date.uid;
-      this.showTooltip();
+      this.#activeUid = this.date.uid;
+      this.#showTooltip();
     }
 
     if (this.hasTooltip) {
-      this.datepickerService.keyDatePopoverStream
-        .pipe(takeUntil(this.ngUnsubscribe))
+      this.#datepickerService.keyDatePopoverStream
+        .pipe(takeUntil(this.#ngUnsubscribe))
         .subscribe((date) => {
           if (date) {
-            this.activeUid = date.uid;
+            this.#activeUid = date.uid;
           } else {
-            this.activeUid = '';
+            this.#activeUid = '';
           }
           // If this day has an open popover and they have moved off of the day close the popover.
-          if (this.date.uid !== this.activeUid) {
-            this.hideTooltip();
+          if (this.date?.uid !== this.#activeUid) {
+            this.#hideTooltip();
           }
         });
     }
   }
 
   public ngOnDestroy(): void {
-    this.ngUnsubscribe.next();
-    this.ngUnsubscribe.complete();
+    this.#ngUnsubscribe.next();
+    this.#ngUnsubscribe.complete();
   }
 
   public onDayMouseenter(): void {
-    this.cancelPopover = false;
+    this.#cancelPopover = false;
     if (this.hasTooltip) {
-      this.showTooltip();
-      this.datepickerService.keyDatePopoverStream.next(this.date);
+      this.#showTooltip();
+      this.#datepickerService.keyDatePopoverStream.next(this.date);
     }
   }
 
   public onDayMouseleave(): void {
-    this.cancelPopover = true;
+    this.#cancelPopover = true;
     if (this.hasTooltip) {
-      this.hideTooltip();
+      this.#hideTooltip();
     }
-    this.datepickerService.keyDatePopoverStream.next(undefined);
+    this.#datepickerService.keyDatePopoverStream.next(undefined);
   }
 
   public onPopoverClosed(): void {
-    this.popoverOpen = false;
+    this.#popoverOpen = false;
   }
 
   public onPopoverOpened(): void {
-    this.popoverOpen = true;
+    this.#popoverOpen = true;
     /* istanbul ignore else */
-    if (this.cancelPopover) {
+    if (this.#cancelPopover) {
       // If the popover gets opened just as a mouseout event happens, close it.
-      this.hideTooltip();
-      this.cancelPopover = false;
+      this.#hideTooltip();
+      this.#cancelPopover = false;
     }
   }
 
   public getKeyDateLabel(): string {
     if (this.hasTooltip) {
-      return this.date.keyDateText.join(', ');
+      // Safety check
+      /* istanbul ignore next */
```

Comment:
Because `getKeyDateLabel` is only ever called from something within an `ngIf=hasTooltip` which check from `keyDateText` not being `undefined` or an empty array. So we need the ternary here for strict mode; however, the `: ''` case will never be hit because of the wrapping code outside of it.

Reply:
Couldn't you adjust the parent condition and simplify all of it into a single ternary and still get 100% coverage?
```
return this.hasTooltip && this.date?.keyDateText
  ? this.date.keyDateText.join(', ')
  : '';
```